main-command/src/main/scala/sbt/State.scala

```diff
@@ -72,12 +72,15 @@ final case class State(
  * @param currentExecId provide the execId extracted from the original State.
  * @param combinedParser the parser extracted from the original State.
  */
+@deprecated("unused", "1.4.2")
 private[sbt] final case class SafeState(
     currentExecId: Option[String],
     combinedParser: Parser[() => sbt.State]
 )
```

Comment:
If it's `private[sbt]`, maybe we could rip it out?

Reply:
We could. This just felt easier than updating mima.

Reply:
Oh, the other thing is that there are public apis in `EvaluateTask` that expose `SafeState` (which sort of seems like a compiler bug) so we'd have to break binary compatibility of `EvaluateTask` which is neither `private[sbt]` nor in the internal package.

Reply:
```
sbt:sbtRoot> mimaReportBinaryIssues
[info] sbt-launch: mimaPreviousArtifacts is empty, not analyzing binary compatibility.
[info] sbtRoot: mimaPreviousArtifacts is empty, not analyzing binary compatibility.
[info] sbt-client: mimaPreviousArtifacts is empty, not analyzing binary compatibility.
[error]  * object sbt.SafeState does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("sbt.SafeState$")
[error]  * class sbt.SafeState does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("sbt.SafeState")
[info] Scripted sbt Redux: mimaPreviousArtifacts is empty, not analyzing binary compatibility.
[info] sbt-dependency-tree: mimaPreviousArtifacts is empty, not analyzing binary compatibility.
[error]  * static method currentlyRunningEngine()java.util.concurrent.atomic.AtomicReference in class sbt.EvaluateTask does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("sbt.EvaluateTask.currentlyRunningEngine")
[error]  * static method lastEvaluatedState()java.util.concurrent.atomic.AtomicReference in class sbt.EvaluateTask does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("sbt.EvaluateTask.lastEvaluatedState")
[error]  * method lastEvaluatedState()java.util.concurrent.atomic.AtomicReference in object sbt.EvaluateTask does not have a correspondent in current version                    
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("sbt.EvaluateTask.lastEvaluatedState")
[error]  * method currentlyRunningEngine()java.util.concurrent.atomic.AtomicReference in object sbt.EvaluateTask does not have a correspondent in current version                
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("sbt.EvaluateTask.currentlyRunningEngine")
[error] stack trace is suppressed; run last mainProj / mimaReportBinaryIssues for the full output
[error] stack trace is suppressed; run last commandProj / mimaReportBinaryIssues for the full output
[error] (mainProj / mimaReportBinaryIssues) Main: Binary compatibility check failed!
[error] (commandProj / mimaReportBinaryIssues) Command: Binary compatibility check failed!
[error] Total time: 43 s, completed Oct 20, 2020 10:10:19 PM
```
I doubt that `SafeState` was widely used but I'm not sure if we should take it out.

Reply:
ok.